src/main/scala/is/hail/methods/VEP.scala

```diff
@@ -308,74 +311,119 @@ object VEP {
-                            val allele_num = xsplit(alleleNumIndex)
+                        rvb.startArray(tr_aa.length)
+                        tr_aa.foreach { x =>
+                          val xsplit = x.split("\\|")
+                          val allele_num = xsplit(alleleNumIndex)
+                          rvb.addString(
                             if (allele_num.isEmpty)
                               x
                             else
                               xsplit
                                 .updated(alleleNumIndex, alleleMap.getOrElse(xsplit(alleleNumIndex).toInt, "").toString)
-                                .mkString("|")
-                        }: IndexedSeq[Annotation])
-                      } else
-                        (v, tr_aa: IndexedSeq[Annotation])
+                                .mkString("|"))
+                        }
+                        rvb.endArray()
+                      } else {
+                        rvb.startArray(tr_aa.length)
+                        tr_aa.foreach(rvb.addString)
+                        rvb.endArray()
+                      }
                     case None =>
                       warn(s"No VEP annotation found for variant $v. VEP returned $s.")
-                      (v, IndexedSeq.empty[Annotation])
+                      rvb.setMissing()
                   }
+                  rvb.endStruct()
+                  rv2.set(region, rvb.end())                  
                 } else {
-                  val a = JSONAnnotationImpex.importAnnotation(JsonMethods.parse(s), vepSignature)
-                  val vvep = variantFromInput(inputQuery(a).asInstanceOf[String])
-
+                  rvb.start(t)
+                  
+                  val vvep = variantFromInput(s)
                   if (!nonStarToOriginalVariant.contains(vvep))
                     fatal(s"VEP output variant ${ vvep } not found in original variants.\nVEP output: $s")
 
                   val v = nonStarToOriginalVariant(vvep)
+                  
+                  rvb.startStruct()
+                  rvb.addAnnotation(localVariantType, v)
+                  
+                  // a = JSONAnnotationImpex.importAnnotation(JsonMethods.parse(s), vepSignature)
+                                    
                   if (vvep != v) {
-                    val alleleMap = getNonStarToOriginalAlleleIdxMap(v)
-                    (v, consequenceIndices.foldLeft(a.asInstanceOf[Row]) {
-                      case (r, i) =>
-                        if (r(i) == null)
-                          r
-                        else {
-                          r.update(i, r(i).asInstanceOf[IndexedSeq[Row]].map {
-                            x => x.update(0, alleleMap.getOrElse(x.getInt(0), null))
-                          })
-                        }
-                    })
+                    fatal("not yet implemented") // FIXME how to handle with region values?
```

Comment:
Don't.  Convert to region values later.

I've never quite been happy with this code.  You can talk to Laurent about how it's used if you want to improve it.  Basically, I think we should include another annotation that maps the original alleles to the non-star alleles and then users can remap the VEP annotations and/or access them through the map.